### PR TITLE
couchdb 3.4.1

### PR DESCRIPTION
**Pull request by cho-m**
Documentation / Release notes are out https://docs.couchdb.org/en/stable/; however, https://couchdb.apache.org only lists 3.3.3 so may want to wait a bit.

### CODE REVIEW

File: Formula/c/couchdb.rb [hunk 1]
```diff
@@ -46,24 +38,16 @@ class Couchdb < Formula
...
-    spidermonkey = Formula["spidermonkey@91"]
-    inreplace "configure", '[ ! -d "/usr/local/include/${SM_HEADERS}" ]',
-                           "[ ! -d \"#{spidermonkey.opt_include}/${SM_HEADERS}\" ]"
-    inreplace "src/couch/rebar.config.script" do |s|
-      s.gsub! "-I/usr/local/include/mozjs", "-I#{spidermonkey.opt_include}/mozjs"
-      s.gsub! "-L/usr/local/lib", "-L#{spidermonkey.opt_lib} -L#{HOMEBREW_PREFIX}/lib"
-    end
-
-    system "./configure", "--spidermonkey-version", spidermonkey.version.major.to_s
+    system "./configure", "--disable-spidermonkey", "--js-engine=quickjs"
```
Comment (carlocab) [inline Formula/c/couchdb.rb:41]: No `quickjs` dependency?

Comment (cho-m) [reply Formula/c/couchdb.rb:37]: It didn't seem possible to unbundle.

`quickjs` is at least designed to be embedded and only builds static libs so there is less of an advantage as dependency.

**Review by carlocab — APPROVED** [review]
I don't mind switching to a pre-release if it means we can un-deprecate this sooner. CC @Homebrew/core for thoughts.

File: Formula/c/couchdb.rb [hunk 2]
```diff
@@ -46,24 +37,16 @@ class Couchdb < Formula
...
     # setting new database dir
     inreplace "rel/couchdb/etc/default.ini", "./data", "#{var}/couchdb/data"
     # remove windows startup script
-    rm_r("rel/couchdb/bin/couchdb.cmd")
+    rm("rel/couchdb/bin/couchdb.cmd")
     # install files
     prefix.install Dir["rel/couchdb/*"]
-    if File.exist?(prefix/"Library/LaunchDaemons/org.apache.couchdb.plist")
-      (prefix/"Library/LaunchDaemons/org.apache.couchdb.plist").delete
+    if (prefix/"Library/LaunchDaemons/org.apache.couchdb.plist").exist?
```
Comment (MikeMcQuaid) [inline Formula/c/couchdb.rb:48]: Doesn't this mean it'll always exist or not exit for a given version of `couchdb`?

**Review by MikeMcQuaid — APPROVED** [review]
Agreed, happy to ship a prerelease over deprecating.

File: Formula/c/couchdb.rb (hunk 2 above)
Comment (cho-m) [reply Formula/c/couchdb.rb:48]: I only modified style but haven't checked usage. I am guessing either it is only created on macOS or the removal logic is outdated.

**Comment by cho-m** [comment]
I think the reason why this release isn't on homepage (i.e. has no binaries) is https://github.com/apache/couchdb/blob/main/src/docs/src/whatsnew/3.4.rst#version-340
> CouchDB version 3.4.0 includes a feature to automatically upgrade password hashes to a newer algorithm and a configuration option that enables this feature by default ...
> ...
> The issue was found after the formal 3.4.0 release process has concluded, so the **source release is available normally, but the CouchDB team has not made 3.4.0 convenience binaries available**. The team recommends to upgrade to 3.4.1 instead when it is available.

File: Formula/c/couchdb.rb (hunk 2 above)
Comment (cho-m) [reply Formula/c/couchdb.rb:48]: May have been removed in v2: https://github.com/apache/couchdb/commit/a19f7f70bf83bb2403f990668a5827c6775d617c

Will see if possible to just get rid of this.

**Review by carlocab — APPROVED** [review]

**Comment by cho-m** [comment]
Officially released https://blog.couchdb.org/2024/09/27/3-4-1/

**Pull request merged**